docs/content/api-overview/resources/databricks-workspace.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "Databricks Workspace"
+date: 2020-12-13T15:43:30+00:00
+chapter: false
+weight: 8
+---
+
+#### Overview
+The Databricks Workspace builder is used to create Azure Databricks Workspaces
+
+* Workspace (`Microsoft.Databricks/workspaces`)
+
+#### Builder Keywords
+| Keyword | Purpose |
+|-|-|
+| name | Sets the name of the workspace. |
+| pricing_tier | Sets the pricing tier of the workspace. Defaults to Standard Tier |
+| enable_encryption | Turns on workspace encryption encryption. |
+| key_vault | Sets the name of the key vault where the encryption key is stored |
```

Comment:
Would you be able to give me some more information on what ResourceId and ResourceName are used for and where/when each of them should be used. I'm not sure I fully understand them.

Reply:
Sure.

A `ResourceName` represents the raw name of a resource (as the name suggests ;-)). In F# it's common to try to avoid working with raw strings to avoid accidentally mixing two strings that mean semantically different things - thing of it is a wrapper around a string.

A `ResourceId` is a type that contains the full *path* to a resource - so it contains the name, but also optionally the resource group and the full path to the resource. We'll be adding subscription id to this at some point soon, too.

Reply:
If this is confusing, I'm happy to clean this up myself and you can see the diff of what I've done as a result.

Reply:
Thanks for offering to clean it up. I've made an attempt at using ResourceRef and ResourceName. Hopefully I've used them as intended. The operation should now take a KeyVaultConfig, an Arm.KeyVault.Vault or a string as the name.